### PR TITLE
Persist the search term between refreshes

### PR DESCRIPTION
**Pull request by OrDuan**
Saves the last searched text in local storage so when the user refresh the page(auto reload or manually) - the search text appears and shows the results.

Fixes:
- [992](https://github.com/facebook/react-devtools/issues/992)
- [295](https://github.com/facebook/react-devtools/issues/295)

Would be great to get feedback about:
- Is using the local storage is a bad practice?
- When should the text be cleared from the local storage?(if ever)

### CODE REVIEW

File: frontend/SettingsPane.js [hunk 1]
```diff
@@ -50,6 +50,11 @@ class SettingsPane extends React.Component {
     const doc = ReactDOM.findDOMNode(this).ownerDocument;
     // capture=true is needed to prevent chrome devtools console popping up
     doc.addEventListener('keydown', this._key, true);
+
+    const lastSearch = localStorage.getItem('searchText') || '';
+    if (lastSearch !== '') {
+      this.props.onChangeSearch(lastSearch);
```
Comment (bvaughn) [inline frontend/SettingsPane.js]: I don't think this is the best place to do this 😄 We should re-initialize this value in `Store` so we can avoid rendering multiple times

**Comment by OrDuan** [comment]
@bvaughn thanks for the feedback!
I updated the code to the the storage utility. 
I can not find an easy way to init the `Store` with the local storage data, I tried the contractor and the `_mountComponent` both called too early, while the  input's text has the last searched text, it doesn't trigger the search itself to reveal the nodes.

Where would you write the logic that fire the store's `changeSearch`?

**Comment by bvaughn** [comment]
When the search text changes, [`SettingsPane` calls `props.onChangeSearch`](https://github.com/facebook/react-devtools/blob/e863aad2a4a049c59f754a53b16cab6cfa68e5fb/frontend/SettingsPane.js#L133) which is [connected to `store.changeSearch`](https://github.com/facebook/react-devtools/blob/e863aad2a4a049c59f754a53b16cab6cfa68e5fb/frontend/SettingsPane.js#L176).

So my initial thought would be to check in the `Store` constructor, and if a previous saved search exists, auto-filter by calling `changeSearch`– something like:

```js
const previousSearchText = storage.get(SEARCH_TEXT, ""); // Defaults to ""
if (previousSearchText !== "") {
  this.changeSearch(previousSearchText);
}
```

I also suggest using a constant (e.g. `SEARCH_TEXT`) where you get and set the text rather than an inline string, just to better keep them in sync.

**Comment by bvaughn** [comment]
Thanks for elaborating.

I think we should try to dig in further and find out _why_ it didn't work. It still feels a bit gross to me to do this in a component's did-mount hook via a prop callback.

**Comment by bvaughn** [comment]
React DevTools has been rewritten and [recently launched a new version 4 UI](https://reactjs.org/blog/2019/08/15/new-react-devtools.html) 🎉 The source code for this rewrite was done in a separate repository and is [being merged into the main React repo](https://github.com/facebook/react/pull/16381).

**Thank you for taking the time to contribute to this extension** 🙇‍♂️

Because this PR is for the version code base, I am closing it. If you'd like to contribute to the new extension, please visit [the new repository](https://github.com/facebook/react/issues?q=is%3Aissue+is%3Aopen+label%3A%22Component%3A+Developer+Tools%22).

**Pull request closed without merging**